### PR TITLE
Make install take bensos_install_path as a second parameter

### PR DESCRIPTION
**Pull request by rolandtritsch**
We are currently working to add `asdf` as an additional way to install `benthos` (because this gives us an elegant way to have multiple versions of benthos installed at the same time). #1568

The implementation is [here](https://github.com/benthosdev/benthos-asdf/pull/1).

This PR suggests a small backward compatible change/enhancement so that `install` can be used to implement the `asdf` plugin.

The change is to add the ability to also pass the `install_path` as an optional second parameter.

### CODE REVIEW

File: website/static/sh/install [hunk 1]
```diff
@@ -45,12 +63,12 @@ install_benthos()
 	check_tools
 
 	if [[ -n "$PREFIX" ]]; then
-		install_path="$PREFIX/bin"
+		benthos_install_path="$PREFIX/bin"
```
Comment (rolandtritsch) [inline website/static/sh/install:66]: Just renaming `install_path` to `benthos_install_path` (for consistency).

File: website/static/sh/install [hunk 2]
```diff
@@ -111,19 +129,6 @@ install_benthos()
 	echo "Downloading Benthos for ${benthos_os}/${benthos_arch}${benthos_arm}..."
 	benthos_file="benthos_${benthos_os}_${benthos_arch}${benthos_arm}${benthos_dl_ext}"
 
-	if [[ "$#" -eq 0 ]]; then
```
Comment (rolandtritsch) [inline website/static/sh/install:114]: This piece was moved to the top of the file.

File: website/static/sh/install [hunk 3]
```diff
@@ -137,12 +142,15 @@ install_benthos()
...
 
-	echo "Putting benthos in $install_path (may require password)"
-	$sudo_cmd mv "$PREFIX/tmp/$benthos_bin" "$install_path/$benthos_bin"
+	echo "Putting benthos in $benthos_install_path (may require password)"
+	$sudo_cmd mv "$PREFIX/tmp/$benthos_bin" "$benthos_install_path/$benthos_bin"
 	$sudo_cmd rm -- "$dl"
 
 	# check installation
-	$benthos_bin -version
+	$benthos_install_path/$benthos_bin -version
```
Comment (rolandtritsch) [inline website/static/sh/install:150]: When installing with `asdf` benthos is not on the path yet (this happens when you run `asdf global ...`.

Means we need to distinguish between we can execute the bin and it's presence on the path.

**Comment by Jeffail** [comment]
👍 thanks @rolandtritsch, the whitespace needs fixing as it's all over the place but I'm happy with the changes.

**Comment by rolandtritsch** [comment]
> +1 thanks @rolandtritsch, the whitespace needs fixing as it's all over the place but I'm happy with the changes.

Uuupppsss. Sorry. Fixed.

**Comment by Jeffail** [comment]
> > +1 thanks @rolandtritsch, the whitespace needs fixing as it's all over the place but I'm happy with the changes.
> 
> Uuupppsss. Sorry. Fixed.

It's consistent but can I be a total pain and ask for tabs instead of spaces? 😅 I don't have a strong preference either way but it makes for a smaller diff since we had tabs before.

**Comment by rolandtritsch** [comment]
> It's consistent but can I be a total pain and ask for tabs instead of spaces? sweat_smile I don't have a strong preference either way but it makes for a smaller diff since we had tabs before.

Good point. Fixed.

**Review by Jeffail — APPROVED** [review]
LGTM, thanks @rolandtritsch!

**Comment by rolandtritsch** [comment]
Hi @Jeffail. I now want to cleanup the implementation of the [asdf][] plugin, but when I run `curl -Lsf https://sh.benthos.dev` I am still seeing the old/previous `install.sh`. When will it be available on `sh.benthos.dev`? (cc: @mihaitodor)

[asdf]: https://github.com/benthosdev/benthos-asdf/pull/1

**Comment by mihaitodor** [comment]
Hey @rolandtritsch, Happy New Year! I think that gets updated when a new release is published, but I'll have to defer to @Jeffail for that.

**Pull request merged**